src/main/java/org/elasticsearch/cluster/metadata/MetaDataCreateIndexService.java

```diff
@@ -554,11 +555,37 @@ public int compare(IndexTemplateMetaData o1, IndexTemplateMetaData o2) {
 
     private void validate(CreateIndexClusterStateUpdateRequest request, ClusterState state) throws ElasticsearchException {
         validateIndexName(request.index(), state);
-        String customPath = request.settings().get(IndexMetaData.SETTING_DATA_PATH, null);
+        validateIndexSettings(request.index(), request.settings());
+    }
+
+    public void validateIndexSettings(String indexName, Settings settings) throws ElasticsearchException {
+        String customPath = settings.get(IndexMetaData.SETTING_DATA_PATH, null);
         if (customPath != null && nodeEnv.isCustomPathsEnabled() == false) {
-            throw new IndexCreationException(new Index(request.index()),
-                    new ElasticsearchIllegalArgumentException("custom data_paths for indices is disabled"));
+            throw new IndexCreationException(new Index(indexName),
```

Comment:
Would it make sense to include this error into the validationErrors list as well?